main.go

```diff
@@ -199,6 +199,10 @@ func main() { //nolint:funlen
 		setupLog.Error(err, "unable to update from legacy operator version")
 	}
 
+	if err = upgrade.RemoveResource(mgr.GetConfig(), dscApplicationsNamespace); err != nil {
```

Comment:
For now we only want to remove one resouce, let's inline

```suggestion
	if err = upgrade.RemoveResource(mgr.GetConfig(), dscApplicationsNamespace, gvr.ODHApplication, "jupyter"); err != nil {
```

Reply:
the main reason i did not want to pass gvr and CR name directly in `RemoveResource` is that it is a bit different as how we did for `UpdateFromLegacyVersion` 
a very clean call in main.go then the detail is in upgrade.go
but if to have `RemoveResource` in main.go for a very short live for one release, then i do not think it is a problem.